code/game/atoms_movable.dm

```diff
@@ -529,6 +529,7 @@
 		throw_impact(get_turf(src), speed)
 	if(loc)
 		stop_throw()
+		SEND_SIGNAL(loc, COMSIG_TURF_LANDED, src)
```

Comment:
It's not strictly limited to open turfs, although in practice it probably will be. And it's not tied to the turf's throw_impact, since that isn't called if the thrown object hits anything else in the turf.